bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java

```diff
@@ -212,4 +215,14 @@ public String toString() {
         return DateFormat.getDateTimeInstance().format(time) + ": " + name + " -> " + state.toString();
     }
 
+    @Override
+    @DynamoDBAttribute(attributeName = ATTRIBUTE_NAME_EXPIRATION)
+    public Long getExpiration() {
```

Comment:
I think you need to have getters and setters in subclasses (`DynamoDBBigDecimalItem` & `DynamoDBStringItem`), not in the parent class in order AWS SDK reflection to pick it up correctly. At least previously I encountered clear issues otherwise.

In any case, you can run the integration tests against real DynamoDB, see the instructions in [`.test` bundle `README`](https://github.com/openhab/openhab1-addons/blob/master/bundles/persistence/org.openhab.persistence.dynamodb.test/README.md)

Reply:
How about renaming it to more verbose `expirationDateEpochSeconds` to remove ambiguity with epoch-seconds and epoch-milliseconds?